File: Front/src/api/listaEspera.ts
import { api } from "./axios";

export interface EntradaListaEspera {
  id: string;
  nombreCliente: string;
  telefonoCliente: string;
  emailCliente?: string;
  fechaPreferida?: string;
  estado: "Esperando" | "Notificado" | "Confirmado" | "Expirado";
  fechaCreacion: string;
  fechaNotificacion?: string;
  servicioNombre?: string;
  empleadoNombre?: string;
}

export const listaEsperaApi = {
  obtener: (estado?: string) =>
    api.get<EntradaListaEspera[]>("/lista-espera", { params: { estado } }).then((r) => r.data),

  cambiarEstado: (id: string, estado: string) =>
    api.patch(`/lista-espera/${id}/estado`, { estado }).then((r) => r.data),

  eliminar: (id: string) => api.delete(`/lista-espera/${id}`),
};

export const listaEsperaPublicoApi = {
  unirse: (data: {
    slug: string;
    nombreCliente: string;
    telefonoCliente: string;
    emailCliente?: string;
    servicioId?: string;
    empleadoId?: string;
    fechaPreferida?: string;
  }) => api.post("/publico/lista-espera", data).then((r) => r.data),
};

File: Back/AppointVaAPI/AppointVaAPI/Models/ListaEspera.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AppointVaAPI.Models
{
    public class ListaEspera
    {
        [Key]
        public Guid Id { get; set; }

        [Required]
        public Guid NegocioId { get; set; }
        [ForeignKey("NegocioId")]
        public Negocio? Negocio { get; set; }

        public Guid? ServicioId { get; set; }
        [ForeignKey("ServicioId")]
        public Servicio? Servicio { get; set; }

        public Guid? EmpleadoId { get; set; }
        [ForeignKey("EmpleadoId")]
        public Empleado? Empleado { get; set; }

        [Required]
        public string NombreCliente { get; set; } = string.Empty;

        [Required]
        public string TelefonoCliente { get; set; } = string.Empty;

        public string? EmailCliente { get; set; }

        public DateTime? FechaPreferida { get; set; }

        [Required, MaxLength(20)]
        public string Estado { get; set; } = "Esperando"; // Esperando | Notificado | Confirmado | Expirado

        [Required]
        public DateTime FechaCreacion { get; set; }

        public DateTime? FechaNotificacion { get; set; }
    }
}
